staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/nop_converter.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conversion
+
+import (
+	"fmt"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// nopConverter is a converter that only sets the apiVersion fields, but does not real conversion. It supports fields selectors.
+type nopConverter struct {
+	clusterScoped bool
+	crd           *apiextensions.CustomResourceDefinition
+}
+
+var _ runtime.ObjectConvertor = &nopConverter{}
+
+func (c *nopConverter) ConvertFieldLabel(version, kind, label, value string) (string, string, error) {
+	// We currently only support metadata.namespace and metadata.name.
+	switch {
+	case label == "metadata.name":
+		return label, value, nil
+	case !c.clusterScoped && label == "metadata.namespace":
+		return label, value, nil
+	default:
+		return "", "", fmt.Errorf("field label not supported: %s", label)
+	}
+}
+
+func (c *nopConverter) Convert(in, out, context interface{}) error {
+	unstructIn, ok := in.(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("input type %T in not valid for unstructured conversion", in)
+	}
+
+	unstructOut, ok := out.(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("output type %T in not valid for unstructured conversion", out)
+	}
+
+	unstructOut.SetUnstructuredContent(unstructIn.UnstructuredContent())
+
+	_, err := c.ConvertToVersion(unstructOut, unstructOut.GroupVersionKind().GroupVersion())
```

Comment:
is looks wrong: we cannot expect that the out object has a group version set, can we?

Reply:
We just set the content of the out from in object. if In object has gv set, the out would have it too, right?

Reply:
Isn't this a noop then if input and output GV matches?

Reply:
> Isn't this a noop then if input and output GV matches?

yes, that's a bug. any info we expect to get from unstructOut must be retrieved before we stomp it with `unstructOut.SetUnstructuredContent(unstructIn.UnstructuredContent())`

if we can't determine the target group version, we have to return an error

at least in some cases, the context arg is used to pass version info, though I'm not sure how normative that is:

https://github.com/kubernetes/kubernetes/blob/d2952c0b2eb018bb360fe92456b439f754e71859/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go#L131-L133

Reply:
That line is the only one calling `ObjectConverter.Convert` (outside of parameter codecs). So we know pretty well what the context is there.